Library/Homebrew/extend/os/linux/extend/ENV/super.rb

```diff
@@ -28,4 +41,74 @@ def determine_dynamic_linker_path(formula)
       "#{HOMEBREW_PREFIX}/lib/ld.so"
     end
   end
+
+  # @private
+  def homebrew_extra_pkg_config_paths
+    paths = []
+    if x11?
+      libs = xorg_recursive_deps.map(&:lib)
+      shares = xorg_recursive_deps.map(&:share)
+      libs.each do |l|
+        paths << l/"pkgconfig"
+      end
+      shares.each do |s|
+        paths << s/"pkgconfig"
+      end
+    end
+    paths
+  end
+
+  def homebrew_extra_aclocal_paths
+    paths = []
+    if x11?
+      shares = xorg_recursive_deps.map(&:share)
+      shares.each do |s|
+        paths << s/"aclocal"
+      end
+    end
+    paths
+  end
+
+  def xorg_include_paths
+    xorg_recursive_deps.map(&:include).map(&:to_s)
+  end
+
+  def xorg_lib_paths
+    xorg_recursive_deps.map(&:lib).map(&:to_s)
+  end
+
+  def homebrew_extra_isystem_paths
+    paths = []
+    paths += xorg_include_paths if x11?
+    paths
+  end
+
+  def homebrew_extra_library_paths
+    paths = []
+    paths += xorg_lib_paths if x11?
+    paths
+  end
+
+  def homebrew_extra_cmake_include_paths
+    paths = []
+    paths += xorg_include_paths if x11?
+    paths
+  end
+
+  def homebrew_extra_cmake_library_paths
+    paths = []
+    paths += xorg_lib_paths if x11?
+    paths
+  end
+
+  def set_x11_env_if_installed
+    ENV.x11 = xorg_installed?
```

Comment:
This is for `brew sh` and it mirrors similar code in extend/os/mac/extend/ENV/super.rb.